spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/TraceWebClientBeanPostProcessor.java

```diff
@@ -274,6 +276,7 @@ public void onNext(ClientResponse response) {
 								.build());
 			}
 			finally {
+				// TODO: is there a way to read the request at response time?
```

Comment:
@rstoyanchev @bclozel  is there any way to do this at the moment or some future release of reactor-netty? The intent is to see the request that actually caused the response. The original request may have been rewritten due to redirects etc.

Reply:
updated the copy/paste :P

Reply:
That would be an `ExchangeFilterFunction` which this [processor is using](https://github.com/spring-cloud/spring-cloud-sleuth/blob/0227c050985ca656306e50eb8ab0c69792b89420/spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/TraceWebClientBeanPostProcessor.java#L139-L149). What you do from there, extending `Mono<ClientResponse>` and wrapping its `Subscriber` is not an expected way to use the API. Nevertheless the request seems to be passed into the former but not into the latter, but probably could be.

Reply:
thanks for the hint! added it to the todo bucket :) https://github.com/spring-cloud/spring-cloud-sleuth/pull/1573/files#diff-bb63d22a058c70307dc3b4e017e8b0b4R98